File: convert.php
<?php

function convertCode($code, $fromLang, $toLang) {
    $url = 'https://carbon.now.sh/api/convert';
    $data = [
        'code' => $code,
        'from' => $fromLang,
        'to' => $toLang
    ];

    $options = [
        'http' => [
            'method' => 'POST',
            'header' => 'Content-Type: application/json',
            'content' => json_encode($data)
        ]
    ];

    $context = stream_context_create($options);
    $response = file_get_contents($url, false, $context);

    if ($response !== false) {
        $result = json_decode($response, true);
        $convertedCode = $result['code'];
        file_put_contents("output.$toLang", $convertedCode);
        echo "Arquivo 'output.$toLang' criado com sucesso.\n";
    } else {
        echo "Erro ao converter o código.\n";
    }
}

function getLanguages() {
    $url = 'https://carbon.now.sh/api/languages';
    $response = file_get_contents($url);

    if ($response !== false) {
        $result = json_decode($response, true);
        $languages = $result['languages'];
        echo "Linguagens disponíveis:\n";
        foreach ($languages as $index => $lang) {
            echo ($index + 1) . ". $lang\n";
        }
        echo count($languages) + 1 . ". Todas as Linguagens\n";
        promptConversion($languages);
    } else {
        echo "Erro ao obter a lista de linguagens.\n";
    }
}

function promptConversion($languages) {
    $inputType = readline("Digite o código ou o caminho do arquivo:\n1. Código\n2. Caminho do arquivo\nOpção: ");

    if ($inputType === '1') {
        $code = readline("Digite o código: ");
        selectLanguages($code, $languages);
    } elseif ($inputType === '2') {
        $filePath = readline("Digite o caminho do arquivo: ");
        $code = file_get_contents($filePath);
        if ($code !== false) {
            selectLanguages($code, $languages);
        } else {
            echo "Erro ao ler o arquivo.\n";
        }
    } else {
        echo "Opção inválida.\n";
    }
}

function selectLanguages($code, $languages) {
    echo "Linguagens disponíveis:\n";
    foreach ($languages as $index => $lang) {
        echo ($index + 1) . ". $lang\n";
    }
    echo count($languages) + 1 . ". Todas as Linguagens\n";

    $fromLangIndex = readline("Digite o número da linguagem de origem: ");
    if ($fromLangIndex < 1 || $fromLangIndex > count($languages) + 1) {
        echo "Opção inválida.\n";
        return;
    }

    $fromLang = ($fromLangIndex === count($languages) + 1) ? 'auto' : $languages[$fromLangIndex - 1];

    $toLangIndex = readline("Digite o número da linguagem de destino: ");
    if ($toLangIndex < 1 || $toLangIndex > count($languages) + 1) {
        echo "Opção inválida.\n";
        return;
    }

    $toLang = ($toLangIndex === count($languages) + 1) ? $languages : [$languages[$toLangIndex - 1]];

    convertCode($code, $fromLang, $toLang);
}

getLanguages();

File: convert.py
import requests

def convert_code(code, from_lang, to_lang):
    url = "https://carbon.now.sh/api/convert"
    data = {
        "code": code,
        "from": from_lang,
        "to": to_lang
    }
    response = requests.post(url, json=data)
    if response.ok:
        result = response.json()
        converted_code = result['code']
        return converted_code
    else:
        print("Erro ao converter o código.")
        return None

def get_languages():
    url = "https://carbon.now.sh/api/languages"
    response = requests.get(url)
    if response.ok:
        result = response.json()
        languages = result['languages']
        return languages
    else:
        print("Erro ao obter a lista de linguagens.")
        return None

def main():
    print("Digite o código ou o caminho do arquivo:")
    input_type = input("1. Código\n2. Caminho do arquivo\nOpção: ")

    if input_type == '1':
        code = input("Digite o código: ")
    elif input_type == '2':
        file_path = input("Digite o caminho do arquivo: ")
        with open(file_path, 'r') as file:
            code = file.read()
    else:
        print("Opção inválida.")
        return

    languages = get_languages()
    if not languages:
        return

    print("Linguagens disponíveis:")
    for index, lang in enumerate(languages, start=1):
        print(f"{index}. {lang}")
    print(f"{len(languages)+1}. Todas as Linguagens")

    from_lang_index = int(input("Digite o número da linguagem de origem: "))
    if from_lang_index < 1 or from_lang_index > len(languages) + 1:
        print("Opção inválida.")
        return
    elif from_lang_index == len(languages) + 1:
        from_lang = "auto"
    else:
        from_lang = languages[from_lang_index - 1]

    to_lang_index = int(input("Digite o número da linguagem de destino: "))
    if to_lang_index < 1 or to_lang_index > len(languages) + 1:
        print("Opção inválida.")
        return
    elif to_lang_index == len(languages) + 1:
        to_lang = languages
    else:
        to_lang = [languages[to_lang_index - 1]]

    for lang in to_lang:
        converted_code = convert_code(code, from_lang, lang)
        if converted_code:
            with open(f'output.{lang}', 'w') as file:
                file.write(converted_code)
            print(f"Arquivo 'output.{lang}' criado com sucesso.")

if __name__ == '__main__':
    main()
